### DISCUSSION TITLE
How to run a jbang script from a private gitlab repository

### DISCUSSION THREAD

**Discussion by basti128** [jbangdev/jbang#1651] [Q&A]
I’m in a bit of a predicament. I got some code that can be run via jbang in a private gitlab repository. Is there a possibility to run that without cloning it by hand with something like : `jbang -token = xxx https://gitlab.com/basti128/test/-/blob/main/src/main/java/ex/example/Test.java` so basically is it possible to transmit some kind of authentication so that jbang is able to access that repository. The reason would be to allow a third party to run some code, but only if they are authorized. If that's possible (Something like that got mentioned in release 0.99: JBang now fully support proxies and credentials to use with private/internal repositories) can you please provide an example on the needed setup/conditions.

**Comment by quintesse** [comment]
I know we have support for `GITHUB_TOKEN` as an environment variable, but there's not (yet) anything like that for GitLab or BitBucket.

There's also support for `JBANG_AUTH_BASIC_USERNAME` and `JBANG_AUTH_BASIC_PASSWORD` (although I guess https://USER:PASSWORD@gitlab.com/etc/etc/etc would work as well).

Other authentication mechanisms are not implemented (yet).

Adding support for GItLab/BitBucket tokens would probably not be too hard (assuming they have something similar to what GitHub does), so contributions are welcome :-)

**Reply by basti128** [reply]
Thx for your suggestions. But unfortunately none of them work, also tried using an access token with JBANG_AUTH_BASIC_PASSWORD, but haven't had any luck there either :grimacing:

**Reply by quintesse** [reply]
No, that'll only work for basic authentication, you can't use it for tokens. But the basic authentication should definitely work, if GitLab supports that of course.
